homeassistant/components/netatmo/sensor.py

```diff
@@ -145,6 +161,41 @@ def get_entities():
 
     async_add_entities(await hass.async_add_executor_job(get_entities), True)
 
+    @callback
+    def add_public_entities():
+        """Retrieve Netatmo public weather entities."""
+        entities = []
+        for area in entry.options.get(CONF_WEATHER_AREAS, {}).values():
+            data = NetatmoPublicData(
+                auth,
+                lat_ne=area[CONF_LAT_NE],
+                lon_ne=area[CONF_LON_NE],
+                lat_sw=area[CONF_LAT_SW],
+                lon_sw=area[CONF_LON_SW],
+            )
+            for sensor_type in SUPPORTED_PUBLIC_SENSOR_TYPES:
+                entities.append(NetatmoPublicSensor(area, data, sensor_type,))
+
+        for device in async_entries_for_config_entry(device_registry, entry.entry_id):
+            if device.model == "Public Weather stations":
+                device_registry.async_remove_device(device.id)
+
+        if entities:
+            async_add_entities(entities)
```

Comment:
Can you end up adding certain areas twice?

Reply:
Not by name, but I am not checking if the selected area already exists. The same area can be added multiple times, which can make sense since it can return average or max values for the available stations.

Once options are comited and in another run the same name is entered it will edit the existing area by that name.
Although if the area is just created in that flow without being submitted it would be overwritten.

Reply:
I meant that could you end up passing certain entity objects twice to `async_add_entities`

Reply:
No, should not happen.